pandas/core/internals/managers.py

```diff
@@ -1235,14 +1235,24 @@ def value_getitem(placement):
                 if len(val_locs) == len(blk.mgr_locs):
                     removed_blknos.append(blkno_l)
                 else:
-                    nb = blk.delete(blk_locs)
+                    nbs = blk.delete(blk_locs)
+                    # Add first block where old block was and remaining blocks at
+                    # the end to avoid updating all block numbers
+                    first_nb = nbs[0]
+                    nbs_tup = tuple(nb for nb in nbs[1:])
```

Comment:
Nit: `tuple(nbs[1:])` to avoid iteration cost?

Reply:
Thx, fixed